plugin/core/transports.py

```diff
@@ -90,14 +90,22 @@ def _read_loop(self) -> None:
             while self._reader:
                 headers = http.client.parse_headers(self._reader)  # type: ignore
                 body = self._reader.read(int(headers.get("Content-Length")))
-                callback_object = self._callback_object()
-                if callback_object:
-                    try:
-                        callback_object.on_payload(_decode(body))
-                    except Exception as ex:
-                        exception_log("Error handling payload", ex)
-                else:
-                    break
+                try:
+                    payload = _decode(body)
+                except Exception as ex:
+                    exception_log("JSON decode error", ex)
+                    continue
+
+                def invoker(p: Dict[str, Any]) -> Callable[[], None]:
+
+                    def invoke() -> None:
+                        callback_object = self._callback_object()
+                        if callback_object:
+                            callback_object.on_payload(p)
+
+                    return invoke
+
+                sublime.set_timeout_async(invoker(payload))
```

Comment:
Just a potentially easier to read suggestion:
```
from functools import partial
```

```suggestion
                def invoke(p: Dict[str, Any]) -> None:
                    callback_object = self._callback_object()
                    if callback_object:
                        callback_object.on_payload(p)

                sublime.set_timeout_async(partial(invoke, payload))
```
I think this should work, not 100% sure.